### PR TITLE
Improvements to line coverage analysis.

### PR DESCRIPTION
**Pull request by mum4k**
- calculate coverage for all directories, not just `src/lib`, this will be future proof when we add more code directories.
- increase the coverage threshold according to current level.
- split `apt` execution to two commands, installing `docker.io` in a separate execution. This allows installation of the remaining packages to continue on machines that cannot install `docker.io`.
- publish the coverage html report as an artifact in the CI pipeline.
- Rename coverage CI steps from `test` to `coverage`.
- fix a typo and reformat the script.

### CODE REVIEW

File: salvo/tools/coverage.sh [hunk 1]
```diff
@@ -51,8 +62,11 @@ sed -e 's/SF.*\.runfiles\/salvo\/\(.*\)$/SF:\1/' -i coverage/salvo.dat
 # Generate HTML coverage report.
 mkdir -p coverage/html
 genhtml coverage/salvo.dat -o coverage/html
+zip -r coverage/html.zip coverage/html/
+
 echo "HTML coverage report generated, view by running:"
-echo "  (cd coverage/html && python3 -m http.server)"
+echo "  (cd salvo/coverage/html && python3 -m http.server)"
+echo "Or download the zip file from salvo/coverage/html.zip"
```
Comment (gyohuangxin) [inline salvo/tools/coverage.sh:69]: Is this information still useful for developers? They can just see the log but cannot perform such operations.

**Review by fei-deng — APPROVED** [review]

File: .azure-pipelines/pipelines.yml [hunk 2]
```diff
@@ -46,16 +46,20 @@ stages:
...
       matrix:
-        test:
+        coverage:
           CI_TARGET: "coverage"
     timeoutInMinutes: 120
     steps:
     - template: bazel.yml
       parameters:
         ciTarget: $(CI_TARGET)
+    - task: PublishPipelineArtifact@1
```
[the remark this answers is not in the document]
Comment (mum4k) [reply .azure-pipelines/pipelines.yml:62]: Good advise, done in 9eb723179f9e363dd85c9c26b22b792698533681.

File: salvo/tools/coverage.sh (hunk 1 above)
Comment (mum4k) [reply salvo/tools/coverage.sh]: I think this is still useful for local executions. The `ci/do_ci.sh` script can also be executed locally to run tests, debug, etc.

I have updated the text in 1f3f2466a226e346a80e8d57b7664c0d1e1ef989 to clarify with regards to the two execution modes.

**Review by gyohuangxin — APPROVED** [review]
LGTM

**Pull request merged**